presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
I am not sure if I understand the semantics Hives privilege model here.
We are only revoking if `existingPrivileges.containsAll(parsePrivilege(requestedPrivilege))`.
What with the cases when `parsePrivilege(requestedPrivilege)` is not contained in `existingPrivilages`, but there is non-empty intersection of the two?
Shouldn't we do sth in that case?